starts out FlowBuilder

da3mon opened the pull request:
**do not merge**

@bryanduxbury could use some ideas for testing. current thoughts: add to the existing PumpTest. Making a separate test file is duplicates a lot from there for a simple delegating-builder.

also, any thoughts on dropping/deprecating the static get() method in favor of always using the new FlowBuilder interface?

Commit:
adds FlowBuilder - fluent interface for FlowDefs.

cleans up the interface so that users only have to know about the
builder on the helper and less about FlowDefs and their interaction with
the CascadingHelper.

bryanduxbury commented on src/main/java/com/squareup/cascading_helpers/FlowBuilder.java:
```diff
@@ -0,0 +1,104 @@
+  public FlowBuilder source(String name, Tap tap) {
+    Pump pump = Pump.prime(name);
+    emittedClasses.addAll(pump.getEmittedClasses());
+    flowDef.addTailSink(pump.toPipe(), tap);
```
you don't want a source pipe to be added as a tail sink. 

you shouldn't need to get emitted classes for a source pipe, since the classes emitted by source pipes will contribute emitted classes to their tail pipes.

da3mon replied:
whoops. that was a copy-pasta error on the source -> tailSink bit.

i'll remove the emittedClasses adding on the sources, too.

bryanduxbury commented:
Approach LGTM. 

Seems like testing will be a pain. We might not have any choice but to build a few flows that exercise all the options and test that they work?

I'm pro on deprecating the old CascadingHelper.get() stuff, so long as we're ready to migrate usages first.

Commit:
adds documentation

Commit:
chains prev Pump through PipeAdapterPump

we need to preserve the getEmittedClasses result from the previous pipe
through the PipeAdapterPump, so that simple things like retain don't
drop off the known classes

Commit:
adds happy path test

between this and the (eventually converter TestPump) there should be
enough coverage and the classes are simple enough that it's well
covered.

also moves TestPump to a more junit 4 style
